app/code/Magento/CustomerImportExport/Model/Import/Customer.php

```diff
@@ -569,11 +569,11 @@ public function getEntityTable()
      */
     public function getValidColumnNames()
     {
-        $this->validColumnNames = array_merge(
-            $this->validColumnNames,
-            $this->customerFields
+        return array_unique(
+            array_merge(
+                $this->validColumnNames,
+                $this->customerFields
```

Comment:
Sorry but I do not really understand. Where is the complexity here? in the `array_unique`? If so, we could remove it as the main problem was simply that the resulting array was saved over and over again into `$this->validColumnNames`. That way, just returning the array_merge will fix that.

Regarding your solution, it will be ok here but in the `CustomerComposite.php` we have to flip 4 arrays. If I understand it correctly that would be:

```
return array_keys(
             array_flip($this->validColumnNames) +
             array_flip($this->_customerAttributes) +
             array_flip($this->_addressAttributes) +
             array_flip($this->_customerEntity->getValidColumnNames())
 );
```

I find this a bit more complicated to understand than my current solution. Does it really make a difference in terms of performance? If so, I would not hesitate to edit my pull request.